lib/js/src/manager/screen/menu/_MenuReplaceUtilities.js

```diff
@@ -0,0 +1,489 @@
+     * @param {MenuCell[]} toCells - The new menu cells
+     */
+    static transferCellIdsFromCells (fromCells = null, toCells = null) {
+        if (fromCells === null || toCells === null || fromCells.length === 0 || fromCells.length !== toCells.length) {
+            return;
+        }
+        toCells.forEach((newCell, index) => {
+            newCell._setCellId(fromCells[index]._getCellId());
+            // Update parent ids
+            if (!newCell.isSubMenuCell()) {
+                return;
+            }
+
+            newCell.getSubCells().forEach(subCell => {
+                subCell._setParentCellId(newCell._getCellId());
+            });
+        });
+    }
+
+    /**
+     * Moves cell listeners from old to new menu cells
+     * @param {MenuCell[]} fromCells - The old menu cells
+     * @param {MenuCell[]} toCells - The new menu cells
+     */
+    static transferCellListenersFromCells (fromCells = null, toCells = null) {
+        if (toCells === null || fromCells === null || fromCells.length === 0 || toCells.length !== fromCells.length) {
+            return;
+        }
+        fromCells.forEach((oldCell, index) => {
+            toCells[index].setMenuSelectionListener(oldCell.getMenuSelectionListener());
+        });
+    }
+
+    /**
+     * Finds artworks in menu cells that need uploading
+     * @param {MenuCell[]} cells - The menu cell list
+     * @param {FileManager} fileManager - Filemanager for checking artwork uploading
+     * @param {WindowCapability} windowCapability - What the HMI is capable of
+     * @param {SdlArtwork[]} uniqueArtworksToUpload - The currently found, unique artworks
+     * @returns {SdlArtwork[]} - The found artworks
+     */
+    static findAllArtworksToBeUploadedFromCells (cells, fileManager = null, windowCapability, uniqueArtworksToUpload = []) {
+        // Make sure we can use images in the menus
+        if (!_ManagerUtility.hasImageFieldOfName(windowCapability, ImageFieldName.cmdIcon)) {
+            return [];
+        }
+
+        cells.forEach(cell => {
+            if (fileManager !== null) {
+                if (fileManager.fileNeedsUpload(cell.getIcon())) {
+                    artworkUniquenessCheck(cell.getIcon());
+                }
+                if (_ManagerUtility.hasImageFieldOfName(windowCapability, ImageFieldName.menuCommandSecondaryImage) && fileManager.fileNeedsUpload(cell.getSecondaryArtwork())) {
+                    artworkUniquenessCheck(cell.getSecondaryArtwork());
+                }
+            }
+            if (cell.isSubMenuCell() && cell.getSubCells().length !== 0) {
+                // pass along the currently found unique artworks in recursive calls to prevent duplicate artworks from subcells
+                uniqueArtworksToUpload = _MenuReplaceUtilities.findAllArtworksToBeUploadedFromCells(cell.getSubCells(), fileManager, windowCapability, uniqueArtworksToUpload);
+            }
+        });
+
+        function artworkUniquenessCheck (incomingArtwork) {
```

Comment:
There's no JS docs for this artworkUniquenessCheck method and it's caught by the linter.